### PR TITLE
Ensure support for Python 3.10 and 3.11 by testing against these versions

### LINKED ISSUE #841: Add tests for python 3.10 and 3.11
Python 3.10 was released over a year ago, 3.11 was released 3mo ago. We need to ensure we support these versions without issues by adding these versions to the nox tests.

### PR DESCRIPTION
**Pull request by jreiberkyle**
**Related Issue(s):**

Closes #841

**Proposed Changes:**

For inclusion in changelog (if applicable):

1. Ensures support for Python 3.10 and 3.11 by testing against these versions.

Not intended for changelog:

1. 

**Diff of User Interface**

Old behavior:
nox only tested against 3.7, 3.8, 3.9

New behavior:
nox tests against 3.7, 3.8, 3.9, 3.10, and 3.11

**PR Checklist:**

- [x] This PR is as small and focused as possible
- [x] If this PR includes proposed changes for inclusion in the changelog, the title of this PR summarizes those changes and is ready for inclusion in the Changelog.
- [-] I have updated docstrings for function changes and docs in the 'docs' folder for user interface / behavior changes
- [-] This PR does not break any examples or I have updated them

**(Optional) @mentions for Notifications:**

### CODE REVIEW

**Review by kevinlacaille — APPROVED** [review]
When I only have Python 3.7-3.9 installed on my machine `nox` fails on 3.10 and doesn’t get to 11. Is this what should happen? Or should it skip those python versions? After installing 3.10-11, all versions (3.7-3.11) passed.

When I get rid of 3.8, here's what `nox`'s output looks like:

```shell
nox > * test-3.7: success
nox > * test-3.8: skipped
nox > * test-3.9: success
nox > * test-3.10: success
nox > * test-3.11: success
```

File: tests/unit/test_io.py [hunk 1]
```diff
@@ -71,4 +71,4 @@ def test_str_to_datetime_success(string, expected):
 
 def test_str_to_datetime_failure():
     with pytest.raises(exceptions.PlanetError):
-        io.str_to_datetime('20210101')
+        io.str_to_datetime('notadate')
```
[the remark this answers is not in the document]
Comment (jreiberkyle) [reply tests/unit/test_io.py:74]: the reason for this change is that '20210101' no longer fails on 3.11 (and therefore the test, which expects a failure, fails). Any ideas for a good 'wrong format' entry?

[the remark this answers is not in the document]
Comment (kevinlacaille) [reply tests/unit/test_io.py:74]: No, I think this is fine! I was unaware that '20210101' failed on 3.11, but great catch!

[the remark this answers is not in the document]
Comment (jreiberkyle) [reply tests/unit/test_io.py:74]: Well, I didn't catch it to be fair. Our tests did. Yay tests!

**Pull request merged**